sklearn/ensemble/_gb.py

```diff
@@ -1070,7 +1084,15 @@ class GradientBoostingClassifier(ClassifierMixin, BaseGradientBoosting):
         relative to the previous iteration.
         ``oob_improvement_[0]`` is the improvement in
         loss of the first stage over the ``init`` estimator.
-        Only available if ``subsample < 1.0``
+        Only available if ``subsample < 1.0``.
+
+    oob_scores_ : ndarray of shape (n_estimators,)
+        The full history of the loss (= deviance) values on the out-of-bag
+        samples. Only available if ``subsample < 1.0``.
+
+    oob_score_ : float
+        The last value of the loss (= deviance) on the out-of-bag samples. It is
+        the same as ``oob_scores_[-1]``. Only available if ``subsample < 1.0``.
```

Comment:
I think we need to clarify losses vs scores here, and only stick to one.

This comment also applies to other doc-strings.

Reply:
Also, `oob_score_` could be removed (IMO) and might not be the same as `oob_scores_[-1]` is `fit` is interrupted.

Reply:
Throughout the API docs we use Loss or Deviance interchangeably, but not scores. There is a `train_score_` attribute which stores loss values on the in-bag samples.
The names for the attributes `oob_score_` and `oob_scores_` was based on the discussion in #23400.

We could rename these attributes to `train_loss`, `oob_loss_` and `oob_final_loss` for consistency.
We already highlight that all these store loss values in the API docs.

Based on the discussion in https://github.com/scikit-learn/scikit-learn/issues/23400#issuecomment-1144977640 and https://github.com/scikit-learn/scikit-learn/issues/23400#issuecomment-1144978648, `oob_score_` was added.

I think we can keep it for convenience, but clarify that it might not always be the same as the `oob_scores_[-1]`. We can add that it is only the same when `n_iter_no_change` is `None`.

Reply:
Thanks for the details.

I think @ogrisel is more authoritative than me regarding choosing a name for this.

Reply:
Thanks for the details.

I think @ogrisel is more authoritative than me regarding choosing a name for this.